### PR TITLE
fix: mark --rulesdir option as deprecated in CLI docs

### PR DESCRIPTION
**Pull request by platinumazure**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: Update docs and CLI option guide to note that `--rulesdir` is deprecated. (It has been considered deprecated for some time but these docs were missing that info.)

#### What changes did you make? (Give an overview)

Update docs and CLI option guide to note that `--rulesdir` is deprecated. (It has been considered deprecated for some time but these docs were missing that info.)

This is effectively a docs change, but commit is tagged as `fix:` due to changing code. The code change is just to reorder options (moving deprecated option to bottom of category) and adding deprecation note to help string.

#### Is there anything you'd like reviewers to focus on?

Any concerns with reordering the options?

Can the CLI help string changes be improved, to maximize both readability and brevity?

### CODE REVIEW

**Comment by platinumazure** [comment]
Created a new gpg key and had some trouble with getting EasyCLA and GitHub commit verification to pass. Should be good now, but please let me know if anything looks suspicious.

File: docs/user-guide/command-line-interface.md [hunk 1]
```diff
@@ -30,62 +30,62 @@ The command line utility has several options. You can view the options by runnin
 eslint [options] file.js [file.js] [dir]
 
 Basic configuration:
-  --no-eslintrc                  Disable use of configuration from .eslintrc.*
-  -c, --config path::String      Use this configuration, overriding .eslintrc.* config options if present
-  --env [String]                 Specify environments
-  --ext [String]                 Specify JavaScript file extensions - default: .js
```
Comment (nzakas) [inline docs/user-guide/command-line-interface.md:36]: Unrelated change seems to have snuck in?

File: docs/user-guide/command-line-interface.md [hunk 2]
```diff
@@ -30,62 +30,62 @@ The command line utility has several options. You can view the options by runnin
...
-  --cache-strategy String        Strategy to use for detecting changed files - either: metadata or content - default: metadata
+  --cache                         Only check changed files - default: false
+  --cache-file path::String       Path to the cache file. Deprecated: use --cache-location - default: .eslintcache
+  --cache-location path::String   Path to the cache file or directory
+  --cache-strategy String         Strategy to use for detecting changed files in the cache - either: metadata or content - default: metadata
 
 Miscellaneous:
-  --init                         Run config initialization wizard - default: false
-  --env-info                     Output execution environment information - default: false
-  --no-error-on-unmatched-pattern  Prevent errors when pattern is unmatched - default: false
```
Comment (nzakas) [inline docs/user-guide/command-line-interface.md:83]: Same?

**Review by nzakas — CHANGES_REQUESTED** [review]
Just a bit of cleanup needed.

File: docs/user-guide/command-line-interface.md (hunk 1 above)
Comment (platinumazure) [reply docs/user-guide/command-line-interface.md:36]: Interesting. What I did here was, I ran `./bin/eslint.js --help` and replaced this entire section with the output, which is the same method I used for previous PRs I submitted of this type. I should have reviewed the diff myself and left a comment explaining this. However, as of right now, `eslint --help` does not include "default: .js" here.

Let me know if you prefer I open a new PR with only a replacement of this help section with the current `eslint --help` output, without the other changes I made intentionally.

File: docs/user-guide/command-line-interface.md (hunk 2 above)
Comment (platinumazure) [reply docs/user-guide/command-line-interface.md:83]: Please see my reply to the other comment. Thanks!

File: docs/user-guide/command-line-interface.md (hunk 1 above)
Comment (nzakas) [reply docs/user-guide/command-line-interface.md:36]: Fair enough. Just wanted to double check.

File: docs/user-guide/command-line-interface.md (hunk 2 above)
Comment (nzakas) [reply docs/user-guide/command-line-interface.md:83]: 👍

**Review by nzakas — APPROVED** [review]
